src/main/java/pl/schoolmanager/controller/TeacherViewController.java

```diff
@@ -74,16 +98,137 @@ public String createSubjectPost(@Valid @ModelAttribute Subject subject, BindingR
 /*		List<Subject> subjects = teacher.getSubject();
 		subjects.add(subject);
 		teacher.setSubject(subjects);
-		m.addAttribute("teacher1", teacher);
 		this.teacherRepository.save(teacher);
 		this.subjectRepository.save(subject);*/
-		
+//		teacher.setId(20l);
+		m.addAttribute("teacher1", teacher);
 		subject.setSchool(school);
-		subject.getTeacher().add(teacher);
+		subject.getTeacher()/*.add(teacher)*/;
+		List<Subject> subjects = teacher.getSubject();
+		subjects.add(subject);
+		teacher.setSubject(subjects);
+		this.teacherRepository.save(teacher);
 		this.subjectRepository.save(subject);
 		return "test";
 	}
 	
+	// READ
+	@GetMapping("/viewSubject/{subjectId}")
+	public String viewSubject(Model m, @PathVariable long subjectId) {
+		HttpSession s = SessionManager.session();
+		Subject subject = this.subjectRepository.findOne(subjectId);
+		s.setAttribute("subject", subject);
+		return "teacher_view/show_subject";
+	}
+
+	// UPDATE
+	@GetMapping("/updateSubject/{subjectId}")
+	public String updateSubject(Model m, @PathVariable long subjectId) {
```

Comment:
i would remove all such comments while method name is already saying what it does